core/email_api.php

```diff
@@ -1395,3 +1395,51 @@ function email_build_visible_bug_data( $p_user_id, $p_bug_id, $p_message_id ) {
 
 	return $t_bug_data;
 }
+
+/**
+ * Registers shutdown function to send queued emails after script execution
+ * @return void
+ */
+function email_shutdown_function_register() {
+	$t_register = false;
+	if( php_sapi_name() == 'cli' ) {
+		# Skip processing for predefined scripts
+		switch( basename( $_SERVER['SCRIPT_NAME'] ) ) {
+			case 'send_emails.php':
+				break;
+			default:
+				$t_register = true;
+		}
+	} else {
+		# Define shutdown function registration time with an arbitrary delay
+		# to prevent concurrent calls when multiple, parallel http requests
+		# are sent by a single user action
+		$t_now = time();
+		$t_registration_time = session_get_int( 'email_shutdown_function_set', $t_now) ;
+
+		if( $t_registration_time <= $t_now  ) {
+			$t_register = true;
+			# 5 seconds delay until the next registration
+			session_set( 'email_shutdown_function_set', $t_now + 5 );
```

Comment:
If I run two pages that generate email in 5 seconds, could this result in the second email being queued up?  I would rather use a model where the pages that can trigger in parallel would skip the registration of the shutdown method if that is possible.